### PR TITLE
add confirmation route to renew-adult-child flow

### PR DESCRIPTION
**Pull request by Fbasham**
### Description
adds `renew/$id/adult-child/confirmation`

### Related Azure Boards Work Items
[AB#4586](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/4586)

### Checklist

- [x] I have tested the changes locally
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`

### CODE REVIEW

File: frontend/app/mappers/benefit-renewal-service-mappers.server.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export interface ToBenefitRenewRequestFromRenewAdultChildStateArgs {
   dentalBenefits: DentalFederalBenefitsState & DentalProvincialTerritorialBenefitsState;
   dentalInsurance: boolean;
   partnerInformation: PartnerInformationState | undefined;
-  contactInformation: ContactInformationState;
+  contactInformation?: ContactInformationState;
```
Comment (nicholas-ly) [inline frontend/app/mappers/benefit-renewal-service-mappers.server.ts:62]: Shouldn't we expect `contactInformation` to be required when mapping to the renew request? 

The applicant should have answered `isNewOrUpdatedPhoneNumber` and `isNewOrUpdatedEmail` at the very least.

File: frontend/app/routes/public/renew/$id/adult-child/confirmation.tsx [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+export async function loader({ context: { configProvider, serviceProvider, session }, params, request }: LoaderFunctionArgs) {
+  const state = loadRenewAdultChildState({ params, request, session });
+  const t = await getFixedT(request, handle.i18nNamespaces);
+  const locale = getLocale(request);
+
+  // prettier-ignore
+  if (state.applicantInformation === undefined ||
+    state.dentalBenefits === undefined ||
+    state.dentalInsurance === undefined ||
+    (state.hasAddressChanged && state.addressInformation===undefined) ||
```
Comment (nicholas-ly) [inline frontend/app/routes/public/renew/$id/adult-child/confirmation.tsx:45]:
```suggestion
    (state.hasAddressChanged && state.addressInformation === undefined) ||
```

File: frontend/public/locales/en/renew-adult-child.json [hunk 3]
```diff
@@ -409,5 +409,59 @@
...
+  },
+  "confirm": {
+    "page-title": "Your renewal for the Canadian Dental Care Plan is complete!",
+    "keep-copy": "Keep a copy of your application",
+    "print-copy-text": "You will <strong>not</strong> receive a confirmation email. It is important to keep a copy of your application. <noPrint>You can use this button to print.<noPrint>",
+    "print-btn": "Print",
+    "whats-next": "What happens next",
+    "begin-process": "We will begin processing your renewal application immediately. You can continue to use your Canadian Dental Care Plan benefits.",
+    "cdcp-checker": "You can use the <cdcpLink>Canadian Dental Care Plan Status Checker</cdcpLink> or call <noWrap>1-833-537-4342</noWrap> to track the progress of your renewal application or to receive more information. You will receive a letter within two weeks of submitting your renewal application to let you know if you're eligible for the upcoming benefit year.",
+    "register-msca-title": "Register for a My Service Canada Account (MSCA) - Coming soon",
```
Comment (nicholas-ly) [inline frontend/public/locales/en/renew-adult-child.json:421]:
```suggestion
    "register-msca-title": "Register for a My Service Canada Account
```

File: frontend/public/locales/en/renew-adult-child.json [hunk 4]
```diff
@@ -409,5 +409,59 @@
...
+    "email": "Email",
+    "mailing": "Mailing address",
+    "home": "Home address",
+    "dental-private": "Access to dental insurance or coverage",
+    "dental-public": "Access to government dental benefits",
+    "child-title": "{{childName}} information",
+    "is-parent": "Parent or legal guardian of child",
+    "yes": "Yes",
+    "no": "No",
+    "no-change": "No change",
```
Comment (nicholas-ly) [inline frontend/public/locales/en/renew-adult-child.json:449]: From Figma

```suggestion
    "no-change": "No update",
```

File: frontend/app/mappers/benefit-renewal-service-mappers.server.ts (hunk 1 above)
Comment (Fbasham) [reply frontend/app/mappers/benefit-renewal-service-mappers.server.ts]: right, I removed a few too many conditionals in the route-helper and over-corrected in the mapper.

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**